File: CSharp/SpecStudioExamplesTestsCSharp/Calculator_glue.cs
namespace SpecStudioExamplesTestsCSharp.Calculator
{
    using System;
    using System.Collections.Generic;
    using SpecStudioExamplesTestsCSharp.common;
    using Microsoft.VisualStudio.TestTools.UnitTesting;
    using static Microsoft.VisualStudio.TestTools.UnitTesting.Assert;

    public class Calculator_glue
    {
        const string DNCString = "?DNC?";

        public void Examples_Calculation_Add_two_numbers(List<AdderString> values)
        {
            foreach (var value in values)
            {
                Console.WriteLine(value);
            }
            Assert.Fail("Not implemented: Examples_Calculation_Add_two_numbers");
        }

        public void Examples_Calculation_Convert_F_to_C(List<FandCString> values)
        {
            foreach (var value in values)
            {
                Console.WriteLine(value);
            }
            Assert.Fail("Not implemented: Examples_Calculation_Convert_F_to_C");
        }

        public void Examples_DataType_IDForm(List<ValidValuesString> values)
        {
            foreach (var value in values)
            {
                Console.WriteLine(value);
            }
            Assert.Fail("Not implemented: Examples_DataType_IDForm");
        }

        public void Examples_BusinessRule_Shipping_Cost(List<ShippingString> values)
        {
            foreach (var value in values)
            {
                Console.WriteLine(value);
            }
            Assert.Fail("Not implemented: Examples_BusinessRule_Shipping_Cost");
        }

        public void Examples_BusinessRule_Discount(List<DiscountingString> values)
        {
            foreach (var value in values)
            {
                Console.WriteLine(value);
            }
            Assert.Fail("Not implemented: Examples_BusinessRule_Discount");
        }

        public void Examples_DataType_Percentage(List<ValidValuesString> values)
        {
            foreach (var value in values)
            {
                Console.WriteLine(value);
            }
            Assert.Fail("Not implemented: Examples_DataType_Percentage");
        }

        public void Examples_DataType_Dollar(List<ValidValuesString> values)
        {
            foreach (var value in values)
            {
                Console.WriteLine(value);
            }
            Assert.Fail("Not implemented: Examples_DataType_Dollar");
        }

        public void Examples_DataType_SimpleText(List<ValidValuesString> values)
        {
            foreach (var value in values)
            {
                Console.WriteLine(value);
            }
            Assert.Fail("Not implemented: Examples_DataType_SimpleText");
        }

    }
}

File: Go/ExampleTests/calculator_glue.go
package calculator

import (
	"testing"
	"calculator/common"
)

type CalculatorGlue struct {
	// Add state fields here
}

func NewCalculatorGlue() *CalculatorGlue { return &CalculatorGlue{} }

func (g *CalculatorGlue) ExamplesCalculationAddTwoNumbers(t *testing.T, values []common.AdderString) {
	_ = values
	t.Fatal("Not implemented: ExamplesCalculationAddTwoNumbers")
}

func (g *CalculatorGlue) ExamplesCalculationConvertFToC(t *testing.T, values []common.FandCString) {
	_ = values
	t.Fatal("Not implemented: ExamplesCalculationConvertFToC")
}

func (g *CalculatorGlue) ExamplesDataTypeIDForm(t *testing.T, values []common.ValidValuesString) {
	_ = values
	t.Fatal("Not implemented: ExamplesDataTypeIDForm")
}

func (g *CalculatorGlue) ExamplesBusinessRuleShippingCost(t *testing.T, values []common.ShippingString) {
	_ = values
	t.Fatal("Not implemented: ExamplesBusinessRuleShippingCost")
}

func (g *CalculatorGlue) ExamplesBusinessRuleDiscount(t *testing.T, values []common.DiscountingString) {
	_ = values
	t.Fatal("Not implemented: ExamplesBusinessRuleDiscount")
}

func (g *CalculatorGlue) ExamplesDataTypePercentage(t *testing.T, values []common.ValidValuesString) {
	_ = values
	t.Fatal("Not implemented: ExamplesDataTypePercentage")
}

func (g *CalculatorGlue) ExamplesDataTypeDollar(t *testing.T, values []common.ValidValuesString) {
	_ = values
	t.Fatal("Not implemented: ExamplesDataTypeDollar")
}

func (g *CalculatorGlue) ExamplesDataTypeSimpleText(t *testing.T, values []common.ValidValuesString) {
	_ = values
	t.Fatal("Not implemented: ExamplesDataTypeSimpleText")
}
